File: my-app/src/types/addOn.ts
export type AddOnType = "radio" | "checkbox";


export interface AddOnGroup {
    id: string;
    title: string;
    description: string;
    is_required: boolean;
    min_select: number,
    max_select: number,
    add_on_options: AddOnOption[]
    menu_add_on_group_id: string
}

export interface AddOnOption {
    id: string;
    name: string;
    price: number;
    // add_on_id: string,
    type: AddOnType;
    add_on_group_id: string,
    is_active: boolean;
};


export interface AddOnOptionPayload {
    name: string;
    add_on_group_id: string;
    price: number,
    type?: AddOnType
    is_active: boolean,
};

export interface UpdateAddonOptionPayload {
    id: string;
    payload: AddOnOptionPayload;
};

File: be-golang/internal/response/addon.go
package response

import "github.com/google/uuid"

type AddOnGroupResponse struct {
	ID           uuid.UUID             `json:"id"`
	Title        string                `json:"title"`
	Description  *string               `json:"description,omitempty"`
	IsRequired   bool                  `json:"is_required"`
	MinSelect    int                   `json:"min_select"`
	MaxSelect    int                   `json:"max_select"`
	MenuAddOnGroupId uuid.UUID         `json:"menu_add_on_group_id,omitempty"`
	AddOnOptions []AddOnOptionResponse `json:"add_on_options"`
}

type AddOnOptionResponse struct {
	ID       uuid.UUID `json:"id"`
	Name     string    `json:"name"`
	Price    float64   `json:"price"`
	IsActive bool      `json:"is_active"`
	Type     string    `json:"type"`
}
